core/domain/state_domain_test.py

```diff
@@ -76,7 +95,12 @@ def test_get_all_html_content_strings(self):
             exploration.init_state.interaction.get_all_html_content_strings())
         self.assertEqual(
             html_list,
-            ['<p>Feedback</p>', '<p>list_of_sets_of_html_strings</p>', '', ''])
+            [
```

Comment:
You should also be writing tests for each interaction separately.

I also don't see any tests for the negative cases? Check that other cases with "Equals" are handled correctly, etc. Try to think about all places where things might go wrong and write tests for those.

Reply:
done, Added interactions having html